apps/src/applab/ProtectedDesignWorkspace.jsx

```diff
@@ -0,0 +1,44 @@
+import { ApplabInterfaceMode } from './constants';
+import React from 'react';
+import { connect } from 'react-redux';
+
+const ProtectedDesignWorkspace = React.createClass({
+  propTypes: {
+    interfaceMode: React.PropTypes.oneOf([
+      ApplabInterfaceMode.CODE,
+      ApplabInterfaceMode.DESIGN,
+      ApplabInterfaceMode.DATA
+    ]).isRequired
+  },
+
+  shouldComponentUpdate: function (nextProps) {
+    // This component is currently semi-protected. We don't want to completely
```

Comment:
"Protected" seems ok to me as a prefix since the component is actually protecting its DOM state from being blown away, and I can't think of a way in which a misunderstanding of what this means would lead to a wrong outcome. Hopefully we can get rid of this class soon enough and merge it with `DesignWorkspace` once we are rendering with React rather than with `renderDesignWorkspace()`.